tail/tail_test.go

```diff
@@ -65,13 +65,13 @@ func TestTailFuzz(t *testing.T) {
 			}
 			written = append(written, rec)
 		}
-		time.Sleep(time.Second)
-		cancel()
 	}()
 
 	wr := wal.NewReader(rc)
 
-	for wr.Next() {
+	// Expect `count` records; read them all, if possible. The test will
+	// time out if fewer records show up.
```

Comment:
Wording suggestion: `The test will time out if fewer records show up` -> `The test will time out if fewer records show up AND [timeout](https://golang.org/cmd/go/#hdr-Testing_flags) flag is not disabled.`

An alternative is to add timer within this test case, so this test case does not rely on `timeout` flag.

Reply:
IMO, if someone disables the timer they get what they deserve (they should know what they're doing) :)